lib/spack/spack/modules.py

```diff
@@ -156,18 +151,27 @@ def write(self):
         if not os.path.exists(module_dir):
             mkdirp(module_dir)
 
-        # If there are no paths, no need for a dotkit.
-        if not self.paths:
+        # Environment modifications guessed by inspecting the installation prefix
+        env = inspect_path(self.spec.prefix)
+        # Package-specific environment modifications
+        # FIXME : decide how to distinguish between calls done in the installation and elsewhere
+        env.extend(self.spec.package.environment_modifications(None))
+        # site_specific = ...`
+        if not env:
             return
 
         with open(self.file_name, 'w') as f:
-            self._write(f)
+            self.write_header(f)
+            for line in self.process_environment_command(env):
+                f.write(line)
 
-
-    def _write(self, stream):
-        """To be implemented by subclasses."""
+    def write_header(self, stream):
         raise NotImplementedError()
 
+    def process_environment_command(self, env):
+        for command in env:
+            # FIXME : how should we handle errors here?
```

Comment:
If you really want to get fancy, you could call `inspect.currentframe()` from within `prepend_path`, etc., and store the file/line info from the place where the modification was created.  That would let you tell the user where they screwed up.